Docs: Update JavaScript troubleshooting guide (#24105)
WordPress/gutenberg: docs/designers-developers/developers/tutorials/javascript/troubleshooting.md

## Console Log

The console log is a JavaScript developer's best friend. It is a good practice to work with it open, as it displays errors and notices in one place. See Mozilla's [JavaScript console](https://developer.mozilla.org/en-US/docs/Learn/Common_questions/What_are_browser_developer_tools#The_JavaScript_console) documentation for more.

To open the JavaScript console, find the correct key combination for your broswer and OS:

| Browser | Windows      | Linux        | Mac       |
| ------- | ------------ | ------------ | --------- |
| Firefox | Ctrl+Shift+K | Ctrl+Shift+K | Cmd+Opt+K |
| Chrome  | Ctrl+Shift+J | Ctrl+Shift+J | Cmd+Opt+J |
| Edge    | Ctrl+Shift+J | Ctrl+Shift+J | Cmd+Opt+J |
| Safari  |              |              | Cmd+Opt+C |

### First Step

Your first step in debugging should be to check the JavaScript console for any errors. Here is an example, which shows a syntax error on line 6:

![console error](https://raw.githubusercontent.com/WordPress/gutenberg/master/docs/designers-developers/assets/js-tutorial-console-log-error.png)

### Display your message in console log

> You can also write directly to the console from your JavaScript code, this is useful for debugging and checking values of items. The function used is `console.log`, for example:

Reviewer: ```suggestion
You can also write directly to the console from your JavaScript code for debugging and checking variable values. Use the `console.log` function like so:
```

```js
console.log( 'My message' );
```

Or if you want to include a message and variable, in this case display the contents of settings variable:

```js
console.log( 'Settings value:', settings );
```

### Using console log

> You can also write JavaScript in the console log, this is useful if you want to test a short command. The commands you run apply to the open browser window. Try this example to count how many blocks are in the editor. This uses the [wp.data package](/packages/data/README.md), play with it and try to use the console to browse available functions also.

Reviewer: ```suggestion
You can also write JavaScript directly in the console if you want to test a short command. The commands you run apply to the open browser window. Try this example with the [wp.data package](/packages/data/README.md) to count how many blocks are in the editor. Play with it and also try to use the console to browse available functions.
```

```js
wp.data.select( 'core/block-editor' ).getBlockCount();
```

![JavaScript example command](https://developer.wordpress.org/files/2020/07/js-console-cmd.gif)

> ## Confirm JavaScript is loading

Reviewer: ```suggestion
### Using the `debugger` statement

If you would like to pause code execution at a certain line of code, you can write `debugger;` anywhere in your code. Once the browser sees the statement `debugger;`, it will pause execution of your code. This allows you to inspect all variables around the `debugger`  statement, which is very useful. [See this MDN page for more information](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/debugger).

## Confirm JavaScript is loading
```

thought it might be useful to add a section on the debugger
